src/window.ts

```diff
@@ -98,6 +98,12 @@ export class Window extends Adw.ApplicationWindow {
     this.default_width = note.width;
     this.default_height = note.height;
 
+    this.update_title();
+
+    this.note.connect("title-changed", () => {
```

Comment:
You can also use a binding here, to bind the note's title to the window's title directly